### DISCUSSION TITLE
vcpkg install files appearing in wrong directory

### DISCUSSION THREAD

**Discussion by joculatrix** [microsoft/vcpkg#42627] [Q&A]
Hello!

I've followed through the [Install and use packages with CMake](https://learn.microsoft.com/en-us/vcpkg/get_started/get-started) guide. It worked as-is, but my LSP (clangd) didn't recognize that `fmt/core.h` existed in-editor which is, of course, not great for editing purposes. Based on a suggestion I saw online, I added the following line to my `CMakeLists.txt`:
```
SET(CMAKE_EXPORT_COMPILE_COMMANDS 1)
```
This did get clangd to stop throwing errors, however for some reason doing this caused vcpkg to place `vcpkg_installed/` and `vcpkg-manifest-install.log` in the project root directory instead of under `build/` where those were generated before. Aside from the above line added to my CMakeLists and my generator being set to Unix Makefiles instead of Ninja (I'm on Linux), my project is identical to the example. Anyone know why this happens/how to fix it?

**Comment by Osyotr** [comment]
Setting `CMAKE_EXPORT_COMPILE_COMMANDS` should not affect vcpkg installation. Vcpkg installs to `CMAKE_BINARY_DIR` by default so if you've got `vcpkg_installed` in the project root then either `vcpkg install` was launched from the project root or `VCPKG_INSTALLED_DIR` resolved to it.
Such errors (i.e. first configure ok, subsequent broken) usually happen when CMake cache is messed up, especially if path to cmake toolchain has changed without clearing CMake cache.
